stockroom/external/importer/torchvision_importers.py

```diff
@@ -41,26 +41,61 @@ def variability_status(self):
     def __len__(self):
         return len(self.dataset)
 
+    @classmethod
+    def gen_splits(cls, torchvision_dataset, root):
+        dataset_splits = []
+        for train in [True, False]:
+            dataset = torchvision_dataset(root=root, train=train, download=True)
+            dataset_splits.append(cls(dataset, train))
+        return dataset_splits
+
 
 class Cifar10(TorchvisionCommon):
     name = 'cifar10'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.CIFAR10(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.CIFAR10
+        dataset_splits = super().gen_splits(dataset, root)
+        return dataset_splits
+
+    @staticmethod
+    def _process_data(img, lbl):
+        img = np.transpose(np.array(img), (2, 0, 1))
+        img = np.ascontiguousarray(np.array(img))
+        img = img.astype(np.float32) / 255
+        lbl = np.array(lbl)
+        return img, lbl
 
 
 class Mnist(TorchvisionCommon):
     name = 'mnist'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.MNIST(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.MNIST
+        dataset_splits = super().gen_splits(dataset, root)
+        return dataset_splits
 
 
 class FashionMnist(TorchvisionCommon):
     name = 'fashion_mnist'
 
-    def __init__(self, root, train=True):
-        dataset = datasets.FashionMNIST(root=root, train=train, download=True)
-        super().__init__(dataset, train)
+    @classmethod
+    def gen_splits(cls, root):
+        dataset = datasets.FashionMNIST
+        dataset_splits = super().gen_splits(dataset, root)
+        return dataset_splits
+
+
+class ImageFolder(BaseImporter):
```

Comment:
It might be better to submit as another PR when the feature is complete?

Reply:
sure